modules/govuk_env_sync/files/govuk_env_sync.sh

```diff
@@ -340,14 +340,20 @@ function restore_elasticsearch {
 }
 
 function dump_postgresql {
-  if [ -e "/etc/facter/facts.d/aws_environment.txt" ]; then
-    # We do not need sudo rights to write the output file
-    # shellcheck disable=SC2024
-    sudo pg_dump -U aws_db_admin -h "${database_hostname}" --no-password -F c "${database}" > "${tempdir}/${filename}"
+  pg_server_version=$(sudo psql -U aws_db_admin -h "${database_hostname}" --no-password postgres -c 'show server_version;' --no-align --tuples-only)
+
+  if [[ $pg_server_version =~ ^[0-9]+\.[0-9]+(\.[0-9]+)?$ ]]; then
+    # We're using docker to run pgdump because different versions of postgres need different versions of pgdump.
+    # For example, pgdump for postgres 9 doesn't work for postgres 13 databases.
+    # Docker lets us run whichever version we want, without having to build packages for our Ubuntu version
+    sudo docker run --rm --net=host \
+      -v "${tempdir}:/tmp/" \
+      -v "/root/.pgpass:/tmp/.pgpass" -e PGPASSFILE=/tmp/.pgpass \
```

Comment:
Just for posterity — we chatted on Slack about the use of a constant filename in `/tmp` here (as in [this sort of thing](https://wiki.sei.cmu.edu/confluence/display/c/FIO21-C.+Do+not+create+temporary+files+in+shared+directories)) etc. but agreed that it's moot for now given our general security posture on this thing 😅